projects/hr-portal/src/hr-portal.ts

```diff
@@ -1,83 +1,195 @@
 import { LitElement, css, html } from "lit";
 import { customElement, property } from "lit/decorators.js";
+import "igniteui-webcomponents-grids/grids/combined.js";
+import { configureTheme, defineComponents, IgcAvatarComponent, IgcIconButtonComponent, IgcIconComponent, registerIcon } from "igniteui-webcomponents";
+import { IgcTreeGridComponent, IgcCellTemplateContext } from "igniteui-webcomponents-grids/grids";
+import { icons } from "../public/images/Icons";
+
+
+defineComponents(IgcAvatarComponent, IgcIconComponent, IgcTreeGridComponent, IgcIconButtonComponent);
+configureTheme("fluent");
 
 @customElement("app-hr-portal")
 export default class HrPortal extends LitElement {
-  render() {
-    return html` <div>HR PORTAL</div> `;
+
+  @property({ type: Array })
+  data: Array<any> = [];
+
+  constructor() {
+    super();
+    this.data = [];
+    icons.forEach((icon: { name: string; path: string; category: string }) => {
+      registerIcon(icon.name, icon.path, icon.category);
+    });
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.fetchData();
+  }
+
+  async fetchData() {
+    try {
```

Comment:
Can you move this to a separate service file so that it would be similar as the other samples